docs/data/routes/docs/nextjs/troubleshooting/en.md

```diff
@@ -0,0 +1,156 @@
+
+System Pre-requisites
+Running Node LTS version (use node -v to check)
+
+## App does not build or Home Page shows 404
+
+Use the GraphiQL interface to verify that the sitemap query returns the expected list of page routes.
+
+```
+# YOUR_PATH should be the ID of your site root (home page) in lower-case, with dashes removed
+# YOUR_LANGUAGE should be your default language, as defined in package.json
+
+query{
+  search(where: {
+    AND:[
+      {
+      name:"_path",
+      value:"YOUR_PATH"
+      },
+      {
+      name:"_language",
+      value:"YOUR_LANGUAGE"
+      },
+      {
+      name:"_hasLayout",
+      value :"true"
+      }
+    ]
+  })
+  { 
+    total
+    results {
+        url {
+          path
+        }
+      }
+  }
+}
+```
+
+If you do not see expected results, try the following steps:
+
+1. Populate Solr Schema (Sitecore Control Panel > “Populate Solr Managed Schema”)
+
+2. Rebuild indexes (Sitecore Control Panel > “Indexing Manager”)
+
+## Errors regarding SSL certificates
+
+Error examples
+> Error: unable to verify the first certificate
+>
+> Error: UNABLE_TO_VERIFY_LEAF_SIGNATURE
+
+> UnauthorizedError: invalid signature
+
+If you are working locally, make sure that you have [followed the steps](todo://add_link_to_new_page) for enabling self-signed certificates in Node. Check that you have the Node Extra Certificates path in your system variables.
+
+Node that the steps are different, depending on whether you're using a containerized or on-prem installation of Sitecore.
+
+## Errors when generating GraphQL introspection data
+
+This task is performed if the GraphQL schema changes. It depends on the `scjssconfig.json` file being present and populated.
+
+## Errors deploying a JSS app locally
+
+```
+IMPORT ERROR(S) OCCURRED!
+Exception thrown while importing JSS app
+Exception: Sitecore.Exceptions.AccessDeniedException
+Message: AddFromTemplate - Add access required
+```
+
+This occurs when the import role doesn't have the correct permissions. See the following Knowledge Base article for the solution: https://kb.sitecore.net/articles/650791
+
+## General solution checklist
+
+- Verify that all environment variables defined in the [`.env` file](https://github.com/Sitecore/jss/blob/master/samples/nextjs/.env) file are provided. On local environments, an `.env.local` file should be for this purpose.
+
+- Check that Sitecore has an item for your API key  in the `/sitecore/system/Settings/Services/API Keys` folder and that this item's fields are populated.
+
+- Verify that the "JSS Editing Secret" is set. The`JSS_EDITING_SECRET` [value used by your Next.js app](https://github.com/Sitecore/jss/blob/cb32d3a21b87f488bd4bb5d311d556fd1f8354c4/samples/nextjs/.env#L18) should match the [value used by your Sitecore instance](https://github.com/Sitecore/jss/blob/cb32d3a21b87f488bd4bb5d311d556fd1f8354c4/samples/nextjs/sitecore/config/JssNextWeb.config#L37).
```

Comment:
Cross link to EE doc (/docs/nextjs/experience-editor/walkthrough) for more info?